### ISSUE TITLE
code-gen: export generated query keys for the generated react-query hooks

### ISSUE THREAD

**Issue by dirkdev98** [compasjs/compas#697]
We need to abstract them anyways for cache invalidation as specified in #690 .

If we implement this is will be a 2-step approach. First keep the current keys, expose named functions for the used cache key. Something like

```ts
export function groupNameQueryKey(params: T.GroupNameParams): QueryKey {
  return ['GroupName', params ];
}
```

Once projects have moved to these query key functions we can improve on the query keys by splitting group and name.

```ts
export function groupNameQueryKey(params: T.GroupNameParams): QueryKey {
  return [`group`, `name`, params ];
}
```

This would allow invaliding of all queries starting with `group`. After the first step we can start speccing out #690 and determine the 'final' keys that we may need.

If #690 isn't a good idea, I think we should still do step 1, cause renaming a function is easier than renaming a string, and gives a better typed client.

Some implementation specifics:

- Should be code splittable
- Should be typed
- Should we create a new file, or in the same file?
- We can also 'decorate' the function is that a good idea?

```ts
export function useFooBar(params: T.FooBarParams) {
  return useQuery(useFooBar.queryKey(params), ...);
}

useFooBar.queryKey = function (params: T.FooBarParams): QueryKey {
 return ['FooBar', params]
}
```

/cc @rodymolenaar , @tjonger

**Comment by rodymolenaar** [comment]
This is looking good. I would prefer the decorator approach. Just feels neater to have the queryKey always available where you're using the hook.

I'm all for #690

**Comment by dirkdev98** [comment]
Implemented step 1 in #702 . Should be released in v0.0.117 soon :tm:

**Comment by dirkdev98** [comment]
Improving the query keys will be scoped under #690
